### ISSUE TITLE
Crash only when trying join to server

### ISSUE THREAD

**Issue by gb5hrh6t** [SleepyTrousers/EnderZoo#45]
---- Minecraft Crash Report ----
Description: Ticking screen

java.lang.ClassCastException: crazypants.enderzoo.charge.EntityPrimedCharge cannot be cast to net.minecraft.entity.EntityLivingBase
	at net.minecraft.client.network.NetHandlerPlayClient.func_147281_a(NetHandlerPlayClient.java:743)
	at net.minecraft.network.play.server.S0FPacketSpawnMob.func_148833_a(SourceFile:97)
	at net.minecraft.network.play.server.S0FPacketSpawnMob.func_148833_a(SourceFile:15)
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:212)
	at net.minecraft.client.multiplayer.GuiConnecting.func_73876_c(SourceFile:78)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1650)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:962)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:887)
	at net.minecraft.client.main.Main.main(SourceFile:148)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

**Comment by Suros** [comment]
Can confirm that this issue is happening for me as well. Seems to be a localized event that crashes any clients that load the affected area. Not sure how to find or wipe the faulting object.

**Comment by Suros** [comment]
I think I got it figured out. As the wither cat mobs were causing MFR grinders to have issues, I disabled them. After that, people started to have CTDs which created crash reports the same as the original post had. Later, in an attempt to correct this new issue, I disabled all explosives and concussion creepers. This was followed by a new error in some cases and all the dire wolves began appearing as wither witches.

My theory on this is that disabling a mob type post world gen is causing all the entity IDs to shift and cause invalid data tags to be attached to persisting mobs. Not a programmer but I do specialize in finding the root of problems. Hope this info helps you out!

**Comment by bcg17** [comment]
I also have had this problem ill ask the server owner if he has disabled any of the mobs

Here's my crash report:
http://pastebin.com/SC4NYeAL

**Comment by RoyCurtis** [comment]
Wanted to comment as well; we just disabled Wither Cats on our server. Then a player entered an area and has since kept crashing with the same crash. Had to re-enable wither cats.

As a work around, we are going to try to use ForgeEssentials to deny `fe.protection.mobspawn.forced.enderzoo.WitherCat` and `fe.protection.mobspawn.natural.enderzoo.WitherCat` permissions for _ALL_.

Crash report: https://gist.githubusercontent.com/anonymous/752d83deffc2420b720f/raw/bf4c601bbaed7487b1adedf72ea9ca9b38f32737/crash-2015-02-15_18.51.10-client.txt

**Comment by GizmoTheGreen** [comment]
having this issue as well, I have disabled withercats and witches

**Comment by fguillod** [comment]
This happened to me as well on a Direwolf20 server (pack version 1.7.0, EnderZoo version 1.0.13.30).

I disabled Dire Slimes because they were spawning way too much (I set "B:direSlimeEnabled=false" in EnderZoo.cfg), and after that, some players started to crash when loading specific areas, most certainly where there were Dire Slimes before I disabled them.

To fix it, I re-enabled the Dire Slimes but changed the spawn probability to 0.

**Comment by tterrag1098** [comment]
Because you disabled the dire slime on the server but not on the client. THE CLIENT AND SERVER ENTITY LISTS CANNOT DIFFER.

**Comment by GizmoTheGreen** [comment]
Ooh that makes sense. I'll keep that spawn rate 0 trick for next time.
On Sep 16, 2015 15:34, tterrag1098 <notifications@github.com> wrote:Because you disabled the dire slime on the server but not on the client. THE CLIENT AND SERVER ENTITY LISTS CANNOT DIFFER.

**Comment by starg09** [comment]
@tterrag1098 I just stumbled upon this bug. I disabled direslimes on both server and client a long time ago, and now this happened. No other configs are touched. Will update on what happens.

**Comment by starg09** [comment]
Well, tried reenabling direslimes, and setting the chance to 0. Still, same problem. I even deleted the xmls on both client and server before starting. Sadly, I'll have to disable the mod so my friends don't kill me for testing, sorry 'bout that :(

**Comment by HenryLoenwind** [comment]
I think I'll modify the enable/disable code to do something different. Not registering the entities leads to too many problems.

**Comment by DanielMarquard** [comment]
Just wanted to mention that I found this thread after a day of troubleshooting significant instability for users on our Direwolf20 server, especially when teleporting to Mystcraft dimensions. Last night, we disabled Khndrel Keghts (Dire Slimes), which caused repeated client crashes. I believe re-enabling this resolved the issue, although we also rolled back world files by 18 hours.
